### ISSUE TITLE
ARM longlong return value is wrong

### ISSUE THREAD

**Issue by renevdzee** [NationalSecurityAgency/ghidra#2547]
**Describe the bug**
The order of the r0,r1 pair of registers for returning a long long / uint64_t value is swapped. 

I was implementing the __softfp calling convention (for double values also returned in r0/r1) in the ARM.cspec file and I saw my register order for doubles was swapped compared to 8-byte integer values. Therefor I made a small test for the longlong case and discovered that it was wrong.

**To Reproduce**
Compile the following code

```c
long long ll() {
    return 0x0000000200000001;
}
```
(https://godbolt.org/z/Mc8155)

and decompile the .o file in Ghidra. 

**Expected behavior**

Ghidra should compile this into the original code, however it does not, it decompiles this into:

```c
longlong ll(void)
{
  return 0x100000002;
}
```

**Screenshots**
n/a

**Attachments**
n/a

**Environment (please complete the following information):**
 - OS: Windows 10
 - Java Version: supplied with ghidra
 - Ghidra Version: 9.2

I reversed the order of the pair in ARM.cspec file: https://github.com/renevdzee/ghidra/commit/37677da344b67bf0217eaf5567fcfe96d212b090
(first change, the rest of the commit adds support for the __softfp calling convention)

If you would like me to make a pull request for both, let me know.

**Comment by emteere** [comment]
Thanks for finding the issue and the very complete submission.  We'll fix the order for the next patch release 9.2.1, most likely to be released next week.  We've also added a ticket for reviewing the other joined return values for other processors.

The softfp calling convention wouldn't be able to be accepted.  The joining of registers in the inputs in compiler calling conventions needs some more work to support.  Mostly it works well for register_strategy.  My suggestion is opening another ticket with just the softfp addition

**Comment by ghizard** [comment]
Fixed in Ghidra 9.2.1 release.
